lib/plug/adapters/translator.ex

```diff
@@ -13,14 +13,22 @@ defmodule Plug.Adapters.Translator do
   """
 
   # cowboy 1 format
-  def translate(min_level, :error, :format,
-                {~c"Ranch listener" ++ _, [ref, protocol, pid, reason]}) do
+  def translate(
```

Comment:
Are these cases examples of "Example 2: multi-line function definitions"? Link: https://github.com/elixir-lang/elixir/issues/6643

```elixir
def translate(min_level, :error, :format, message) do
  {~c"Ranch listener" ++ _, [ref, protocol, pid, reason]} = message
  # ...
end
```

Reply:
We need to match on the top, unfortunately.

Reply:
What about?

```elixir
def translate(min_level, :error, :format, data) do
  case data do
    # cowboy 1 format
    {~c"Ranch listener" ++ _, [ref, protocol, pid, reason]} ->
      translate_ranch(min_level, ref, protocol, pid, reason)

    # cowboy 2 format
    {~c"Ranch listener" ++ _, [ref, pid, _stream_id, _stream_pid, reason, _]} ->
      translate_ranch(min_level, ref, :cowboy_protocol, pid, reason)
      
    _ ->
      :none
  end
end

def translate(_min_level, _level, _kind, _data), do: :none
```